packages/clerk-js/src/core/resources/Base.ts

```diff
@@ -67,8 +68,22 @@ export abstract class BaseResource {
     }
 
     if (status >= 400) {
-      throw new ClerkAPIResponseError(statusText, {
-        data: payload?.errors as ClerkAPIErrorJSON[],
+      const errors = payload?.errors as ClerkAPIErrorJSON[];
+      const safeErrorMessage = errors?.[0]?.long_message;
+
+      if (errors?.[0]?.code === 'origin_invalid' && isProductionFromPublishableKey(BaseResource.clerk.publishableKey)) {
+        const prodDomain = BaseResource.clerk.frontendApi.replace('clerk.', '');
+        throw new ClerkAPIResponseError(
+          `Clerk: Production Keys are only allowed for domain "${prodDomain}". \nAPI Error: ${safeErrorMessage}`,
+          {
+            data: errors,
+            status: status,
+          },
+        );
+      }
```

Comment:
Can we move this in a separate file or function, because I'd like us to keep the `_fetch` function as clean as possible?